### PR TITLE
feat(authz): introduce an owner relationship when creating an entity

### PR DESCRIPTION
**Pull request by bobeal**

### CODE REVIEW

File: search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+), entities_more_than_one_admin AS (
+    SELECT entity_id
+    FROM entities
+    WHERE admin_right_count > 1
+), entities_with_oldest_date AS (
+    SELECT entity_id, min(created_at) as created_at
+    FROM temporal_entity_attribute
+    WHERE entity_id IN (select entity_id from entities_more_than_one_admin)
+    GROUP BY entity_id
+), entities_oldest_with_sub AS (
```
Comment (ranim-n) [inline search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql:36]: `entities_with_oldest_sub` not `entities_oldest_with_sub`

File: search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    inner join lateral (
+        select sub
+        from attribute_instance_audit
+        where temporal_entity_attribute = tea.id
+          and time_property = 'CREATED_AT'
+          and sub is not null
+    ) l on true
+    where tea.entity_id = entities_with_oldest_date.entity_id
+    and tea.created_at = entities_with_oldest_date.created_at
+)
```
Comment (ranim-n) [inline search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql:48]: here why did you use `on true` in the inner join ?

File: search-service/src/test/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationServiceTests.kt [hunk 3]
```diff
@@ -177,21 +179,21 @@ class EnabledAuthorizationServiceTests {
...
                 eq(entityId01),
                 emptyList(),
-                listOf(AccessRight.R_CAN_ADMIN)
+                listOf(R_IS_OWNER, R_CAN_ADMIN)
             )
         }
     }
 
     @Test
     fun `it should create admin link for a set of entities`() = runTest {
```
Comment (ranim-n) [inline search-service/src/test/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationServiceTests.kt:190]: the name of the method should be renamed too  :  ` it should create creator link for a set of entities`

File: search-service/src/test/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationServiceTests.kt [hunk 4]
```diff
@@ -421,4 +423,92 @@ class EnabledAuthorizationServiceTests {
             )
         }
     }
+
+    @Test
+    fun `it should returned serialized access control entities with other rigths if user is owner`() = runTest {
```
Comment (ranim-n) [inline search-service/src/test/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationServiceTests.kt:428]: the name of the method : `it should return` and not `it should returned`

File: search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql (hunk 1 above)
Comment (bobeal) [reply search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql:36]: hmm, I am really looking for the oldest sub of each entity

Comment (bobeal) [reply search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql:36]: ah indeed, misread the comment

File: search-service/src/test/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationServiceTests.kt (hunk 3 above)
Comment (bobeal) [reply search-service/src/test/kotlin/com/egm/stellio/search/authorization/EnabledAuthorizationServiceTests.kt:190]: and even `it should create owner link ...`

File: search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql (hunk 2 above)
Comment (bobeal) [reply search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql:48]: indeed, it's not needed here since `sub` cannot be null

**Review by ranim-n — APPROVED** [review]

**Pull request merged**